src/format/KdbxWriter.cpp

```diff
@@ -62,6 +66,16 @@ bool KdbxWriter::writeData(QIODevice* device, const QByteArray& data)
     return true;
 }
 
+void KdbxWriter::extractDatabase(QByteArray& xmlOutput, Database* db)
+{
+    QBuffer buffer;
+    buffer.setBuffer(&xmlOutput);
+    buffer.open(QIODevice::WriteOnly);
+    KdbxXmlWriter writer(m_kdbxVersion);
```

Comment:
I don't like this. It makes assumptions about the child classes. I can see now why you introduced `m_kdbxVersion`, but if you need information from the child classes, add a pure-virtual getter method to the base class. That way it's part of the interface and not some arbitrary implementation detail.